File: src/basics.py
from collections import namedtuple
from enum import Enum

def main():
    list = [1, 2, 3, 4, 5]
    dict = {"one": 1, "two": 2}
    sets = set([2, 3, 4, 2])
    bool = True or False
    str = "nim_is_awesome"
    integer: int = 1
    floats: float = 1.2
    PI = 3.14

    Point = namedtuple('Point', ['x', 'y'])
    p = Point(x=11, y=22)

    class Person:
        def __init__(self, name: str, age: int):
            self.name: str = name
            self.age: str = age
    person = Person(name="James", age=32)

    class Color(Enum):
        RED = 1
        GREEN = 2

if __name__ == '__main__':
    main()

File: src/basics.nim
import tables, sets

proc main() =
  let # Runtime const
    list = @[1, 2, 3, 4, 5] # variable length
    arrays = [1, 2, 3, 4, 5] # fixed size
    dict = {"one": 1, "two": 2}.toTable()
    sets = [2, 3, 4, 2].toSet()
    boolean = true or false
    str = "pycaribbean"
    integer: int = 1
    floats: float = 1.2

  const PI = 3.14 # compile time const

  type
    Point = tuple[x,y: int]
  const p: Point = (x:11, y:22) # mutable

  type
    Person = object
      name: string
      age: int
  var person = Person(name:"James", age:32)

  type
    Color = enum
      RED=1
      GREEN=2

when isMainModule:
  main()
